src/partition/repart.c

```diff
@@ -5556,6 +5552,12 @@ static int context_minimize(Context *context) {
                 log_info("Pre-populating %s filesystem of partition %s twice to calculate minimal partition size",
                          p->format, strna(hint));
 
+                if (!vt) {
+                        r = var_tmp_dir(&vt);
+                        if (r < 0)
+                                return log_error_errno(r, "Could not determine temporary directory: %m");
+                }
+
```

Comment:
This should also be added to the second loop a little further down in this function